api/include/opentelemetry/nostd/list.h

```diff
@@ -0,0 +1,254 @@
+    int remaining_elements = previous_size - counter;
+
+    // Copy the original data that comes before the insertion.
+    std::memcpy(temp, data_, insert_index * sizeof(T));
+    // Copy the inserted data.
+    std::memcpy(temp + insert_index, &element, sizeof(T));
+    // Copy the original data that comes after the insertion.
+    std::memcpy(temp + next_index, data_ + counter, remaining_elements * sizeof(T));
+
+    delete[] data_;
+
+    data_ = temp;
+
+    // Return an iterator at the inserted element.
+    return iterator(data_ + counter);
+  }
+
+  // Insert an element at the end of the list.
+  void push_back(T element)
+  {
+
+    int previous_size = size_;
+    size_++;
+
+    T *temp = new T[size_];
+
+    // Copy the original data into the new array.
+    for (int i = 0; i < previous_size; i++)
+    {
+      temp[i] = data_[i];
+    }
+
+    // Deallocate memory used by previous array.
+    delete[] data_;
+
+    data_ = temp;
+
+    data_[previous_size] = element;
+  };
+
+  // Insert an element at the front of the list.
+  void push_front(T element)
+  {
+    size_++;
+
+    T *temp = new T[size_];
+
+    // Copy the original data into the new array leaving space
+    // for the new element at the front.
+    for (unsigned int i = 1; i < size_; i++)
+    {
+      temp[i] = data_[i - 1];
+    }
+
+    // Deallocate memory used by previous array.
+    delete[] data_;
+
+    data_    = temp;
+    data_[0] = element;
+  };
+
+  // Returns the last element and removes it from the list.
+  T pop_back()
+  {
+    T back_elem = data_[size_ - 1];
+    size_--;
+
+    T *temp = new T[size_];
+
+    // Copy the original data.
+    std::memcpy(temp, data_, size_ * sizeof(T));
+
+    // Deallocate memory used by previous array.
+    delete[] data_;
+
+    data_ = temp;
+
+    return back_elem;
+  }
+
+  // Returns the first element and removes it from the list.
+  T pop_front()
+  {
+    T front_elem = front();
+    size_--;
+
+    T *temp = new T[size_];
+
+    // Copy the original data.
+    std::memcpy(temp, data_ + 1, size_ * sizeof(T));
+
+    // Deallocate memory used by previous array.
+    delete[] data_;
+
+    data_ = temp;
+
+    return front_elem;
+  }
+
+  // Returns the number of elements in the list. //
+  int size() { return size_; }
+
+  list &operator=(const list &other_list)
+  {
+    delete[] data_;
+
+    data_ = new T[other_list.size_];
+
+    memcpy(data_, other_list.data_, other_list.size_ * sizeof(T));
+
+    size_ = other_list.size_;
+
+    return *this;
+  }
+
+  list(const list &other_list)
+  {
+    size_ = other_list.size_;
+
+    data_ = new T[size_];
+
+    memcpy(data_, other_list.data_, other_list.size_ * sizeof(T));
+  }
+
+  // Returns true if the list contents are the same, false if they are not
+  bool operator==(const list &other_list)
+  {
+    if (size_ != other_list.size_)
+    {
+      return false;
+    }
+    
+    return (memcmp(data_, other_list.data_, size_*sizeof(T)) == 0);
```

Comment:
Rather use `std::equal(begin(), end(), other_list.begin())` here.

Reply:
switched to std::equal.